### PR TITLE
[macOS] Release platform thread if no commit was requested for a while

### PR DESCRIPTION
**Pull request by iskakaushik**
Fixes: https://github.com/flutter/flutter/issues/70318

This is a temporary fix. I want to continue investigating why `OnFBO` never gets called with the right frame size in some situations. Given that this is hard to repro and that this fix is being defensive, I wish to land this and continue investigating.

### CODE REVIEW

File: shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.mm [hunk 1]
```diff
@@ -67,10 +67,18 @@ - (void)beginResize:(CGSize)size notify:(dispatch_block_t)notify {
 
   _waiting = YES;
 
-  _condBlockRequestCommit.wait(lock, [&] { return _pendingCommit; });
+  const int wait_time_millis = 500;
```
Comment (stuartmorgan) [inline shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.mm:70]: Please use correct ObjC naming for these variables.

File: shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.mm [hunk 2]
```diff
@@ -67,10 +67,18 @@ - (void)beginResize:(CGSize)size notify:(dispatch_block_t)notify {
...
+  bool request_commit_success = _condBlockRequestCommit.wait_for(
+      lock, std::chrono::milliseconds(wait_time_millis), [&] { return _pendingCommit; });
+  if (!request_commit_success) {
+    NSLog(@"Commit wasn't requested after %d ms, no frame was requested."
```
Comment (stuartmorgan) [inline shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.mm:74]: This seems really bad; please file an issue and add a comment here referencing that issue so it's clear that this code shouldn't exist and where the underlying issue is tracked.

File: shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.mm [hunk 3]
```diff
@@ -67,10 +67,18 @@ - (void)beginResize:(CGSize)size notify:(dispatch_block_t)notify {
...
+          @" Releasing the platform thread.",
+          wait_time_millis);
+  } else {
+    [_delegate resizeSynchronizerFlush:self];
+    [_delegate resizeSynchronizerCommit:self];
```
Comment (stuartmorgan) [inline shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.mm:79]: What are the implications of not calling these?

File: shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.mm [hunk 4]
```diff
@@ -67,10 +67,18 @@ - (void)beginResize:(CGSize)size notify:(dispatch_block_t)notify {
...
+  }
 
-  [_delegate resizeSynchronizerFlush:self];
-  [_delegate resizeSynchronizerCommit:self];
   _pendingCommit = NO;
```
Comment (stuartmorgan) [inline shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.mm:82]: If the frame comes in slightly late, do we risk screwing up the subsequent resize's wait? I'm finding it hard to reason about the potential ripple effects of ignoring the fact that we didn't get the release we were expecting, so this needs *thorough* comments explaining why it's safe.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by knopp** [comment]
I think better approach would be to not block platform thread, until we have rendered at least one frame. I was able to somewhat reproduce the problem, and the issue is that engine requests frame and assumes it gets it, but at that time the `onDrawFrame` callback is not set yet (which the engine has no way of knowing). Normally this isn't an issue, because eventually `runApp` is called, which sets the callback and produces frame, but for some reason with debugger that's not always happening. 

If we postpone the wait until we know the framework in producing frames, the problem should disappear.

**Comment by iskakaushik** [comment]
I agree with @knopp suggestion, and @stuartmorgan's concern that this is a bad idea. I created this hacky fix to identify that the issue we were seeing was indeed because we were blocking the platform thread.

Waiting for the first frame to be produced before we block the platform thread seems like the right direction here.

**Comment by iskakaushik** [comment]
Seems like @knopp is working on a PR for this. Will close this in favor of the future PR.

**Pull request closed without merging**